### DISCUSSION TITLE
Deleted Github Files Want To Recover

### DISCUSSION THREAD

**Discussion by TradingMachineAI** [community/community#78593] [New to GitHub]
### Select Topic Area

Question

### Body

Hi all. I accidentally deleted a lot of my files with a commit and now I want to recover those. Can I do that somehow ? 
![image](https://github.com/community/community/assets/143895603/9c81687d-a2a7-4f71-8ca3-298d47d15d1f)

**Comment by DeadGolden0** [comment]
Hi, if you've deleted files in a Git repository with a commit, you can recover those files. Git has a few different ways to undo changes including the ```git reflog``` and ```git reset``` commands.

Here's a step-by-step guide to recover your files:

**1 - Use the reflog to find the commit before the delete:**

Open your terminal or command prompt and navigate to the repository directory.
Use the command ```git reflog``` to see a list of recent commits. Look for the commit just before the deletion happened.

**2 - Check out the commit:**

Once you've found the right commit, you can check it out to see if all your files are there. Use ```git checkout <commit-hash>``` where <commit-hash> is the hash of the commit before your deletion.

**3 - Create a new branch (optional):**

If everything looks correct, you might want to create a new branch with these files by using ```git checkout -b <new-branch-name>``` so you don't lose the current state of your main branch.

**4 - Reset to the commit before the delete:**

If you want to reset your current branch to the state before you deleted the files, use ```git reset --hard <commit-hash>```. This will make your current branch point to the commit before the deletion.

**Warning**: git reset --hard will discard all changes in the working directory and index since the commit you reset to. This can lead to data loss if you have uncommitted changes, so use it with caution.

Alternatively, if you don't want to change the history, you can use the ```git checkout <commit-hash> -- <file-path>``` command to restore specific files from a previous commit to your current working directory.

Remember to replace <commit-hash> with the actual commit ID and <new-branch-name> with the name you want to give to the new branch. <file-path> should be replaced with the path to the file you want to restore.

If you're not comfortable with these commands or the situation is complex, consider making a backup of the current state of your repository before proceeding.
